sentry-ruby/lib/sentry/faraday.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+module Sentry
+  module Faraday
+    OP_NAME = "http.client"
+    SPAN_ORIGIN = "auto.http.faraday"
+    BREADCRUMB_CATEGORY = "http"
+
+    module Connection
+      # Since there's no way to preconfigure Faraday connections and add our instrumentation
+      # by default, we need to extend the connection constructor and do it there
+      #
+      # @see https://lostisland.github.io/faraday/#/customization/index?id=configuration
+      def new(url = nil, options = nil)
```

Comment:
nit: 
any reason for overriding `new` and `extend`ing rather than overriding `initialize` and `prepend`ing ? just a bit cleaner since most of our patches are `prepend`ing